plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/BannedElement.java

```diff
@@ -70,5 +72,24 @@ String getMarkerId() {
   int getSeverity() {
     return severity;
   }
+  
+  @Override
+  public boolean equals(Object object) {
+    if (object == this) {
+      return true;
+    }
+    if (object == null || !(object instanceof BannedElement)) {
```

Comment:
object instanceof BannedElement is only safe in this context if BannedElement is final because of subclasses